### PR TITLE
Fix/bitwig

### PR DESCRIPTION
**Pull request by mrVanDalo**
###### Motivation for this change

fixing #49530
I needed to split up libxkbcommon to version 8 and version 7.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11157,7 +11157,9 @@ with pkgs;
 
   libxdg_basedir = callPackage ../development/libraries/libxdg-basedir { };
 
-  libxkbcommon = callPackage ../development/libraries/libxkbcommon { };
+  libxkbcommon = libxkbcommon_8;
+  libxkbcommon_8 = callPackage ../development/libraries/libxkbcommon { };
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Introducing `libxkbcommon_8` and aliasing it to `libxkbcommon` shouldn't be done for something as trivial as this.

File: pkgs/development/libraries/libxkbcommon/libxkbcommon_7.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    })
+  ];
+
+  mesonFlags = [
+    "-Denable-wayland=false"
+    "-Dxkb-config-root=${xkeyboard_config}/etc/X11/xkb"
+    "-Dx-locale-root=${libX11.out}/share/X11/locale"
+  ];
+
+  doCheck = false; # fails, needs unicode locale
```
Comment (worldofpeace) [inline pkgs/development/libraries/libxkbcommon/libxkbcommon_7.nix]: Can't reproduce locally.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Why that?

Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix]: @Mic92 Seems unnecessary to me, you disagree?

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: It is quite a common pattern in nixpkgs that we keep have aliases like that even if there are only two versions.

File: pkgs/development/libraries/libxkbcommon/libxkbcommon_7.nix (hunk 2 above)
Comment (mrVanDalo) [reply pkgs/development/libraries/libxkbcommon/libxkbcommon_7.nix]: Yeah this came in after the version bump, according to git history.

**Comment by mrVanDalo** [comment]
as @worldofpeace pointed out there are some build configuration which are not correct, this was due to the fact, I used the build script for version 8 to build version 7. I fixed it by putting the build script of the older version 7 in place (from the git-history).

**Comment by worldofpeace** [comment]
> I fixed it by putting the build script of the older version 7 in place (from the git-history).
Great, shouldn't be any problems then.

Because the Borg has added some higher rebuild labels(501+) this should probably be delivered to `staging`.

**Comment by pbogdan** [comment]
In principle for the purposes stated in the PR this shouldn't need to rebuild anything except `libxkbcommon_7` and `bitwig`.

File: pkgs/development/libraries/libxkbcommon/default.nix [hunk 3]
```diff
@@ -1,7 +1,8 @@
 { stdenv, fetchurl, fetchpatch, meson, ninja, pkgconfig, yacc, xkeyboard_config, libxcb, libX11, doxygen }:
 
 stdenv.mkDerivation rec {
-  name = "libxkbcommon-0.8.0";
+  name = "libxkbcommon-${version}";
+  version = "0.8.0";
```
Comment (Mic92) [inline pkgs/development/libraries/libxkbcommon/default.nix]: this is causing the mass-rebuild.

Comment (Mic92) [reply pkgs/development/libraries/libxkbcommon/default.nix]: If you undo it, we can merge it into master.

Comment (mrVanDalo) [reply pkgs/development/libraries/libxkbcommon/default.nix]: OK

Comment (mrVanDalo) [reply pkgs/development/libraries/libxkbcommon/default.nix]: DONE :+1:

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build libxkbcommon_7

**Pull request merged**